### PR TITLE
Cleanup deps

### PR DESCRIPTION
**Pull request by Goooler**
## Description

- Split versions & Bumping.
- Add & Remove flags

And you can enable [Renovate bot](https://docs.renovatebot.com/) to update deps automatically.

## Type of change

✅ General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by yasanglass** [comment]
About Renovate, I'm not sure how different it is from Dependabot but I have used Dependabot and it becomes spammy and annoying very fast.

File: build.gradle [hunk 1]
```diff
@@ -49,9 +49,6 @@ allprojects {
             tasks.withType(JavaCompile) {
                 options.bootstrapClasspath = files([frameworkJar] + (options.bootstrapClasspath.files as Iterable<File>))
             }
-            tasks.withType(org.jetbrains.kotlin.gradle.internal.KaptWithoutKotlincTask) {
-                it.classpath.from(files([frameworkJar]))
-            }
```
Comment (Goooler) [inline build.gradle:54]: No kapt usages now.

**Comment by Goooler** [comment]
> About Renovate, I'm not sure how different it is from Dependabot but I have used Dependabot and it becomes spammy and annoying very fast.

1. Dependabot can't bump Gradle.
2. Renovate supports version catalogs.

**Comment by yasanglass** [comment]
> 1. Dependabot can't bump Gradle.
> 2. Renovate supports version catalogs.

Idk about version catalogs but I'm not sure what you mean by it cant bump gradle? It bumped everything for me afaik. Anyways, that does not fix the spam issue but we can try I guess.

File: build.gradle (hunk 1 above)
Comment (Goooler) [reply build.gradle:54]:
```
1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':SystemUIShared:compileDebugKotlin'.
> Could not resolve all files for configuration ':SystemUIShared:detachedConfiguration5'.
   > Failed to transform R.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
> Task :iconloaderlib:compileDebugKotlin
      > Execution failed for ClasspathEntrySnapshotTransform: /home/runner/work/lawnchair/lawnchair/SystemUIShared/build/intermediates/compile_r_class_jar/debug/R.jar.
         > Check failed.
   > Failed to transform classes.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for ClasspathEntrySnapshotTransform: /home/runner/work/lawnchair/lawnchair/CompatLib/build/intermediates/compile_library_classes_jar/debug/classes.jar.
         > Check failed.
   > Failed to transform classes.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for ClasspathEntrySnapshotTransform: /home/runner/work/lawnchair/lawnchair/CompatLibVR/build/intermediates/compile_library_classes_jar/debug/classes.jar.
         > Check failed.
   > Failed to transform classes.jar to match attributes {artifactType=classpath-entry-snapshot, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for ClasspathEntrySnapshotTransform: /home/runner/work/lawnchair/lawnchair/CompatLibVS/build/intermediates/compile_library_classes_jar/debug/classes.jar.
         > Check failed.
```

Comment (Goooler) [reply build.gradle:54]: Something failed, just reverted in f42a0c6f338bbdb3f6f0e52d5f86c4097d54b19e.

**Comment by Goooler** [comment]
Fix build after https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui/pull/3 merged.

**Comment by Goooler** [comment]
CC @msfjarvis

**Review by msfjarvis — APPROVED** [review]

**Pull request merged**